File: src/chain/examples/idvalidation/ScanQualityValidator.java
package chain.examples.idvalidation;

public class ScanQualityValidator extends IdValidator {
    @Override protected boolean check(IdDocument doc) {
        return doc.getScanHash() != null && doc.getScanHash().length() >= 8 && doc.isCleanBackground();
    }
    @Override protected String name() { return "ScanQualityValidator"; }
}

File: src/chain/examples/idvalidation/ScanQualityValidator.js
'use strict';

const IdValidator = require('./IdValidator');

class ScanQualityValidator extends IdValidator {
  check(doc) { return !!doc.scanHash && doc.scanHash.length >= 8 && doc.cleanBackground; }
  name()     { return 'ScanQualityValidator'; }
}

module.exports = ScanQualityValidator;
